vhdbuilder/packer/generate-vhd-publishing-info.sh

```diff
@@ -57,6 +57,14 @@ if [ "$sas_token" == "" ]; then
 fi
 vhd_url="${STORAGE_ACCT_BLOB_URL}/${VHD_NAME}?$sas_token"
 
+echo "Testing whether the generated sas token works"
+vhd_size=$(curl -sI $vhd_url | grep -i Content-Length | awk '{print $2}')
```

Comment:
should we just check the response code instead? I think even for a 401/403/404, the response size would be non-zero XML payload?

Reply:
```bash
res=$(curl -o /dev/null -w "%{http_code}" -fsSL -I "$vhd_url")
if test "$res" != "200"; then
  exit 1
fi
```

or similar?

Reply:
I think that it is same. This command gets the content length. I have tested that the invalid token will show empty result but the valid token will get a non-empty `integer`.

Reply:
` curl -sI $vhd_url` with a valid token but a non-exist blob returns below response
```
HTTP/1.1 404 The specified blob does not exist.
Transfer-Encoding: chunked
Server: Windows-Azure-Blob/1.0 Microsoft-HTTPAPI/2.0
x-ms-request-id: 0da001d6-901e-000f-56a3-414af3000000
x-ms-version: 2018-11-09
x-ms-error-code: BlobNotFound
Date: Thu, 16 Feb 2023 01:09:47 GMT
```